client/components/banner/index.jsx

```diff
@@ -128,6 +145,10 @@ export class Banner extends Component {
 			);
 		}
 
+		if ( 'no-icon' === icon ) {
```

Comment:
Would recommend a separate explicit prop like a `noIcon` boolean and then something similar to:

```jsx
// render
{ ! this.props.noIcon && this.getIcon() }
```